pyproject.toml

```diff
@@ -41,10 +41,11 @@ dependencies = [
     "jinja2>=3.0.3",
     "jupyter_core",
     "jupyter-lsp>=1.5.1",
-    "jupyter_server>=1.16.0,<3",
+    "jupyter_server==2.0.0rc3",
```

Comment:
I actually lowered the `jupyter_server` requirement to `>=1.15` as of 0.3.1. However, without the contents manager events emitted by JS2, you'll run into the issue of new file IDs being created on edits.